src/scores/continuous/flip_flop_impl.py

```diff
@@ -348,3 +349,208 @@ def _encompassing_sector_size_np(
     n_unique_angles = (angular_diffs != 0).sum(axis=0)
     result = np.where(n_unique_angles <= 2, np.max(angular_diffs, axis=0), result)
     return result
+
+
+def flip_flop_index_proportion_exceeding(
+    data: xr.DataArray,
+    sampling_dim: str,
+    thresholds: Iterable,
+    is_angular: bool = False,
+    dims: Optional[Sequence[str]] = None,
```

Comment:
Can you use `FlexibleDimensionTypes` ?

Reply:
Can you split the `dims` arg out into `reduce_dims` and `preserve_dims` like other scoring functions in scores?